descqa/shear_test.py

```diff
@@ -292,17 +305,18 @@ def run_on_single_catalog(self, catalog_instance, catalog_name, output_dir):
         fig, ax = plt.subplots(nrows=2, ncols=ntomo, sharex=True, squeeze=False, figsize=(ntomo*5, 5))
         zmeans = np.linspace(self.zlo, self.zhi, ntomo+2)[1:-1]
         for ii in range(ntomo):
+            f1.write(str(ii) +"\n")
             z_mean = zmeans[ii]
             zlo2 = z_mean - self.z_range
             zhi2 = z_mean + self.z_range
             print(zlo2, zhi2)
-            zmask = (catalog_data[self.z] < zhi2)*(catalog_data[self.z] > zlo2)
+            zmask = (catalog_data[self.z]< zhi2)*(catalog_data[self.z] > zlo2)
             # compute shear auto-correlation
             cat_s = treecorr.Catalog(
-                ra=catalog_data[self.ra][zmask],
-                dec=catalog_data[self.dec][zmask],
-                g1=catalog_data[self.e1][zmask] - np.mean(catalog_data[self.e1][zmask]),
-                g2=-(catalog_data[self.e2][zmask] - np.mean(catalog_data[self.e2][zmask])),
+                ra=catalog_data[self.ra][zmask*mask_mag],
+                dec=catalog_data[self.dec][zmask*mask_mag],
+                g1=catalog_data[self.e1][zmask*mask_mag] - np.mean(catalog_data[self.e1][zmask*mask_mag]),
+                g2=-(catalog_data[self.e2][zmask*mask_mag] - np.mean(catalog_data[self.e2][zmask*mask_mag])),
```

Comment:
use ` & ` instead of `*` to join boolean masks. Or better, save `zmask & mask_mag` in a variable in advance.